server/src/test/java/org/elasticsearch/action/admin/ReloadSecureSettingsIT.java

```diff
@@ -0,0 +1,418 @@
+                            for (final NodesReloadSecureSettingsResponse.NodeResponse nodeResponse : nodesReloadResponse.getNodes()) {
+                                assertThat(nodeResponse.reloadException(), notNullValue());
+                                assertThat(nodeResponse.reloadException().getMessage(), containsString("If shouldThrow I throw"));
+                            }
+                        } catch (final AssertionError e) {
+                            reloadSettingsError.set(e);
+                        } finally {
+                            latch.countDown();
+                        }
+                    }
+
+                    @Override
+                    public void onFailure(Exception e) {
+                        reloadSettingsError.set(new AssertionError("Nodes request failed", e));
+                        latch.countDown();
+                    }
+                });
+        latch.await();
+        if (reloadSettingsError.get() != null) {
+            throw reloadSettingsError.get();
+        }
+        // even if one plugin fails to reload (throws Exception), others should be
+        // unperturbed
+        assertThat(mockReloadablePlugin.getReloadCount() - initialReloadCount, equalTo(1));
+        // mock plugin should have been reloaded successfully
+        assertThat(mockReloadablePlugin.getSeedValue(), equalTo(seedValue));
+    }
+
+    public void testReloadWhileKeystoreChanged() throws Exception {
+        final PluginsService pluginsService = internalCluster().getInstance(PluginsService.class);
+        final MockReloadablePlugin mockReloadablePlugin = pluginsService.filterPlugins(MockReloadablePlugin.class)
+                .stream().findFirst().get();
+        final Environment environment = internalCluster().getInstance(Environment.class);
+        final int initialReloadCount = mockReloadablePlugin.getReloadCount();
+        for (int i = 0; i < randomIntBetween(4, 8); i++) {
+            // write keystore
+            final SecureSettings secureSettings = writeEmptyKeystore(environment, new char[0]);
+            // read seed setting value from the test case (not from the node)
+            final String seedValue = KeyStoreWrapper.SEED_SETTING
+                    .get(Settings.builder().put(environment.settings()).setSecureSettings(secureSettings).build())
+                    .toString();
+            // reload call
+            successfulReloadCall();
+            assertThat(mockReloadablePlugin.getSeedValue(), equalTo(seedValue));
+            assertThat(mockReloadablePlugin.getReloadCount() - initialReloadCount, equalTo(i + 1));
+        }
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        final List<Class<? extends Plugin>> plugins = Arrays.asList(MockReloadablePlugin.class, MisbehavingReloadablePlugin.class);
+        // shuffle as reload is called in order
+        Collections.shuffle(plugins);
```

Comment:
you need to use `Collections.shuffle(plugins, random())` to fix the build failure